aws/resource_aws_s3_bucket_test.go

```diff
@@ -1682,6 +1682,48 @@ func TestAccAWSS3Bucket_objectLock(t *testing.T) {
 	})
 }
 
+func TestAccAWSS3Bucket_forceDestroy(t *testing.T) {
```

Comment:
Received a test failure with this new test:

```
--- FAIL: TestAccAWSS3Bucket_forceDestroy (10.65s)
    testing.go:630: Error destroying resource! WARNING: Dangling resources
        may exist. The full state and error is shown below.
        
        Error: errors during apply: error S3 Bucket force_destroy: error deleting at least one object version, last error: InvalidArgument: x-amz-bypass-governance-retention is only applicable to Object Lock enabled buckets.
```

Reply:
Yes, I get that too now.
I must have not retested after modifying `deleteAllS3ObjectVersions()` to bypass governance retention restrictions.
I'll fix.

Reply:
Documented that `force_destroy` should only be set on S3 objects if the bucket has Object Lock enabled and modified the code that calls `deleteAllS3ObjectVersions` to only force destroy if the specified bucket has Object Lock enabled.